pkg/state/transaction_checker.go

```diff
@@ -1202,7 +1202,7 @@ func (tc *transactionChecker) checkDataWithProofs(transaction proto.Transaction,
 		return nil, err
 	}
 	utf16KeyLen := tx.Version == 1 && !isRideV6Activated
-	if err := tx.Entries.Valid(utf16KeyLen); err != nil {
+	if err := tx.Entries.Valid(true, utf16KeyLen); err != nil {
```

Comment:
Shouldn't we also call here `IsProtobuf`? The logic would be the same, but there won't be any hard code

Reply:
Nope, for data transaction that's correct. Empty keys are forbidden for any data transaction version.